torch/_dynamo/variables/builder.py

```diff
@@ -1072,6 +1072,44 @@ def wrap_fx_proxy(tx, proxy, example_value=None, **options):
 
 # Note: Unfortunate split due to some gross classes existing that subclass TensorVariable
 # Should be compositional instead
+#
+# This is a horribly complicated function that does too many things, to
+# explain what it does, let's first talk about the classic usage wrap_fx_proxy
+# for a TensorVariable.  There are two primary modes of use:
```

Comment:
Do you want to move this part up to `wrap_fx_proxy`?

Reply:
No? Comment should be close to the relevant implementation.